src/ring_theory/algebra_tower.lean

```diff
@@ -326,4 +327,33 @@ theorem is_basis.smul {ι : Type v₁} {b : ι → S} {κ : Type w₁} {c : κ 
 by rw [← set.range_smul_range, submodule.span_smul hb.2, ← submodule.restrict_scalars'_top R S A,
     submodule.restrict_scalars'_inj, hc.2]⟩
 
+theorem is_basis.smul_repr
+  {ι κ : Type*} {b : ι → S} {c : κ → A}
+  (hb : is_basis R b) (hc : is_basis S c) (x : A) (ij : ι × κ) :
+  (hb.smul hc).repr x ij = hb.repr (hc.repr x ij.2) ij.1 :=
```

Comment:
Should we have a second version `is_basis.smul_repr` where `ij` is replaced with two variables `i` and `j`?